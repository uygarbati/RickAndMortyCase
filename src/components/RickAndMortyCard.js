import React from 'react'
import './RickAndMortyCard.scss'

export default function RickAndMortyCard ({ id, name, thumbnail, handleClickFn }) {
  function handleCardClick () {
    if (handleClickFn) {
      handleClickFn(id)
    }
  }

  return (
    <div
      className="character-card"
      onClick={handleCardClick}
    >
      <img
        src={thumbnail}
        alt={name}
      />
      <div>
        <span>{name}</span>
      </div>
    </div>
  )
}
