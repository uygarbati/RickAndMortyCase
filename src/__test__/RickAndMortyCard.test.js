import React from 'react'
import RickAndMortyCard from '../components/RickAndMortyCard'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<RickAndMortyCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
