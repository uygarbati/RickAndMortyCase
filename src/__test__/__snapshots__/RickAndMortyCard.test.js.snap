// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="character-card"
  onClick={[Function]}
>
  <img />
  <div>
    <span />
  </div>
</div>
`;
